laddertools/ranking.py

```diff
@@ -111,7 +115,293 @@ def get_default_rating(cls):
         return _RatingELO(1000)
 
 
+class _RatingGlicko:
+
+    _initial_rating = 1500
+
+    # scaling factor used to go from / to glicko to glicko-2 scale.
+    _k = 173.7178
+
+    def __init__(self, r0, RD, std):
+        self.r = r0
+        self.std = std
+        self.RD = RD
+
+    def duplicate(self, r=None, RD=None, std=None):
+        return _RatingGlicko(r0=r or self.r, RD=RD or self.RD, std=std or self.std)
+
+    @property
+    def display_value(self):
+        return round(self.value)
+
+    @property
+    def value(self):
+        return self.r
+
+    @property
+    def mu(self):
+        return (self.r - self._initial_rating) / self._k
+
+    @property
+    def phi(self):
+        return self.RD / self._k
+
+    @classmethod
+    def mu_to_rating(cls, mu):
+        return cls._k * mu + cls._initial_rating
+
+    @classmethod
+    def phi_to_RD(cls, phi):
+        return cls._k * phi
+
+    def __repr__(self):
+        return f"<_RatingGlicko r={self.r:.1f}, RD={self.RD:.1f}, std={self.std:.4f}>"
+
+
+class RankingGlicko(_RankingBase):
+    @staticmethod
+    def compute_new_rating(
+        rating,
+        rating_opponents,
+        outcomes,
+        return_intermediate_ratings=False,
+        tau=0.8,
+        eps=1e-6,
+    ):
+        """Compute new rating for `rating`, given a series of M, where opponent i has rating
+        `rating_opponents[i]` and the outcome is `outcomes[i]`.
+
+        Computation is based on [1].
+
+        Args:
+            rating: instance of _RatingGlicko
+            rating_opponents: list of length M; instances of _RatingGlicko
+            outcomes: list of integers; 1 if `rating` won, 0 if lost.
+
+        Returns:
+            A new `_RatingGlicko` instance; the updated rating.
+
+        References:
+            [1]: https://www.glicko.net/glicko/glicko2.pdf
+        """
+
+        def g(rating):
+            return 1 / sqrt(1 + (3 * rating.phi ** 2) / (pi ** 2))
+
+        def E(rating, other):
+            return 1 / (1 + exp(-g(other) * (rating.mu - other.mu)))
+
+        if len(rating_opponents) == 0:
+            # Shortcut: player did not have any matches, so we only increase
+            # the RD.
+            RD = _RatingGlicko.phi_to_RD(sqrt(rating.phi ** 2 + rating.std ** 2))
```

Comment:
you can use `hypot(rating.phi, rating.std)` instead of `sqrt(rating.phi ** 2 + rating.std ** 2)`

Reply:
Oh cool, I was not aware of that function. Thanks!